Promote IPv4/IPv6 dual-stack support to Beta

bridgetkromhout opened the pull request:
Enhancements issue: https://github.com/kubernetes/enhancements/issues/563
Kubernetes PR: https://github.com/kubernetes/kubernetes/pull/98969

Signed-off-by: Bridget Kromhout <bridget@kromhout.org>

annajung commented:
cc @reylejano

tengqm commented on content/en/docs/concepts/services-networking/dual-stack.md:
```diff
@@ -45,34 +45,30 @@ The following prerequisites are needed in order to utilize IPv4/IPv6 dual-stack
 
 An example of an IPv6 CIDR: `fdXY:IJKL:MNOP:15::/64` (this shows the format but is not a valid address - see [RFC 4193](https://tools.ietf.org/html/rfc4193))
 
+Starting in 1.21, IPv4/IPv6 dual-stack defaults to true. Disable if necessary on kube-apiserver, kube-controller-manager, kubelet, and kube-proxy with `--feature-gates="IPv6DualStack=false"`.
```
```suggestion
Starting in 1.21, IPv4/IPv6 dual-stack defaults to true.
You can disable it when necessary by specifying `--feature-gates="IPv6DualStack=false"`
on the kube-apiserver, kube-controller-manager, kubelet, and kube-proxy command line.
```

tengqm commented:
/retitle Promote IPv4/IPv6 dual-stack support to Beta

Please also update table on the page `content/en/docs/reference/command-line-tools-reference/feature-gates.md`.

bridgetkromhout commented:
Thanks, @tengqm!
> Please also update table on the page content/en/docs/reference/command-line-tools-reference/feature-gates.md.

I updated that table per your request, and also updated the other indicator on the top of the docs page so it is also correct for 1.21.

tengqm commented on content/en/docs/reference/command-line-tools-reference/feature-gates.md:
```diff
@@ -123,7 +123,7 @@ different Kubernetes components.
 | `HPAScaleToZero` | `false` | Alpha | 1.16 | |
 | `HugePageStorageMediumSize` | `false` | Alpha | 1.18 | 1.18 |
 | `HugePageStorageMediumSize` | `true` | Beta | 1.19 | |
-| `IPv6DualStack` | `false` | Alpha | 1.15 | |
```
The above line is supposed to be updated rather than dropped so people know the period in which the feature was Alpha.

```
| `IPv6DualStack` | `false` | Alpha | 1.15 | 1.15 |
```

aojea commented:
lgtm

bridgetkromhout commented:
> just FYI, if we want to add a link to the kubeadm docs too

Hi, @aojea -

I see that PR; I assumed it would be merged on its own - is there any action for me to take here in terms of edits to this Pr? Thanks.

aojea commented:
> ee that PR; I assumed it would be merged on its own - is there any action for me to take here in terms of edits to this Pr? Thanks.

I'm not familiar with the website, so I don't know if we use cross references, so I just wanted to let you know, nothing else :)

bridgetkromhout commented:
@aojea Based on my understanding of the website, I believe the link being added in https://github.com/kubernetes/website/pull/26675 is being placed in the correct location. Thanks for flagging!

sftim commented on content/en/docs/concepts/services-networking/dual-stack.md:
```diff
@@ -45,34 +45,32 @@ The following prerequisites are needed in order to utilize IPv4/IPv6 dual-stack
 
 An example of an IPv6 CIDR: `fdXY:IJKL:MNOP:15::/64` (this shows the format but is not a valid address - see [RFC 4193](https://tools.ietf.org/html/rfc4193))
 
+Starting in 1.21, IPv4/IPv6 dual-stack defaults to true.
```
```suggestion
Starting in 1.21, IPv4/IPv6 dual-stack defaults to enabled.
```

sftim commented on content/en/docs/concepts/services-networking/dual-stack.md:
```diff
@@ -45,34 +45,32 @@ The following prerequisites are needed in order to utilize IPv4/IPv6 dual-stack
+## Configure IPv4/IPv6 dual-stack
 
-To enable IPv4/IPv6 dual-stack, enable the `IPv6DualStack` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) for the relevant components of your cluster, and set dual-stack cluster network assignments:
+To configure IPv4/IPv6 dual-stack, set dual-stack cluster network assignments:
```
When features are beta, we typically still mention that the feature gate needs to be enabled, but tweak the wording. This helps people who are using older versions of Kubernetes spot that they might want to swap to reading documentation appropriate for their version.

sftim reviewed (commented):
Thanks for this PR.

The Markdown looks good to me.
I would however revise the text “If you enable IPv4/IPv6 dual-stack networking for your Kubernetes cluster, the cluster will support the simultaneous assignment of both IPv4 and IPv6 addresses.” in https://kubernetes.io/docs/concepts/services-networking/dual-stack/ - to reflect that you typically don't have to take any action to enable it.

You might also want to reword:
> Enabling IPv4/IPv6 dual-stack on your Kubernetes cluster provides the following features:

bridgetkromhout commented:
Thanks for the review, @sftim! I've edited to reflect your feedback.

sftim commented:
/sig network

sftim reviewed (commented):
Markdown LGTM

@kubernetes/sig-network-pr-reviews any concerns?

aojea commented:
LGTM

sftim commented:
/lgtm

reylejano commented:
/assign @ChandaniM123

bridgetkromhout commented:
Hi, @sftim - thanks for giving an LGTM. I see @ChandaniM123 was assigned 3 days ago but I'm not sure what the next step is to get us the `approved` label. Can you provide guidance? Thanks.

sftim commented:
SIG Docs uses slightly different approval conventions from other SIGs, mostly I think because a merge to _master_ is an update to the live website.
A SIG Docs approver (other than me) can approve this change. If _another_ reviewer, reviewing on behalf of SIG Docs, adds their /lgtm, then _I_ can approve this.

sftim commented:
I'm sure we'll get this fully reviewed well before the release. The deadline for docs to be reviewable is 2021-03-24 so you're getting this in nice and early @bridgetkromhout

annajung commented:
Hi @bridgetkromhout, thanks for following up! 
This lgtm and with review from both sig docs / sig network, I'm happy to approve this!

/approve
